pkg/expression/builtin.go

```diff
@@ -301,6 +307,29 @@ func newBaseBuiltinFuncWithFieldTypes(ctx BuildContext, funcName string, args []
 	return bf, nil
 }
 
+// shouldParseToJSON checks whether the function should parse the argument to JSON.
+// When a string is cast to JSON, it may have two different meanings:
+// 1. Quote the string as a JSON string directly. For example, the `abc` will become `"abc"`.
+// 2. Parse the string into a valid JSON value.
+func shouldParseToJSON(funcName string, argIdx int) bool {
```

Comment:
Do we have a ref doc link on mysql? Or it just the behaviour found by ourself.

Reply:
I think it's found by ourselves :facepalm:.